### PR TITLE
Enhance: N - 1 quenching

### PR DESCRIPTION
**Pull request by carrascomj**
Right now, all metabolites in each quenching group $Q$ are assigned a parameter in $q$ out of a neural network such that concentrations are updated as in 

$$
[C]_Q = \text{softmax}(e^{log([C]_Q) - q_Q}) \sum [C]_Q 
$$

This way, the summation of the concentration remains constant. However, this is wasteful in the number of parameters and the softmax is injective (which may introduce identifiability problems). We can rewrite it as

$$
[C]_{Q[-1]} =  \sum [C]_Q - \sum e^{log([C]_Q) - q_Q}
$$

$$
[C]_{Q - 1} = e^{log([C]_Q) - q_Q}
$$

where q_Q contains one less parameter per quenching group. Of course, this may lead to negative concentrations in $[C]_{Q[-1]}$, so the value is clamped to $1\cdot 10^{-11}$.

### CODE REVIEW

**Comment by carrascomj** [comment]
Setting as draft since I am still evaluating this.

**Comment by carrascomj** [comment]
Closing in favor of #4
